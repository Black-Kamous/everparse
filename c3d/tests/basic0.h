typedef int uint32_t;

typedef uint32_t RNDIS_REQUEST_ID;

#define RNDIS_MAJOR_VERSION 0
#define RNDIS_MINOR_VERSION 0

typedef struct
[[
  everparse::entrypoint,
  everparse::process,
  everparse::parameter(uint32_t MessageBodyLength),
  everparse::where(MessageBodyLength == sizeof(this))
]]
_RNDIS_INITIALIZE_REQUEST {
  RNDIS_REQUEST_ID RequestId;
  uint32_t MajorVersion
      [[everparse::constraint(MajorVersion == RNDIS_MAJOR_VERSION && MajorVersion == 0)]];
  uint32_t MinorVersion
      [[everparse::constraint(MinorVersion == MajorVersion),
        everparse::constraint(MinorVersion == 1)]];
  uint32_t MaxTransferSize
      [[everparse::constraint(MaxTransferSize <= MessageBodyLength)]];
} RNDIS_INITIALIZE_REQUEST, *PRNDIS_INITIALIZE_REQUEST;
