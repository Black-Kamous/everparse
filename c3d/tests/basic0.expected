Processing ../tests/basic0.h
